python/pylibcugraph/pylibcugraph/egonet.pyx

```diff
@@ -0,0 +1,172 @@
+
+    Returns
+    -------
+    A tuple of device arrays containing the sources, destinations, edge_weights
+    and the subgraph_offsets(if there are more than one seeds)
+
+    Examples
+    --------
+    >>> import pylibcugraph, cupy, numpy
+    >>> srcs = cupy.asarray([0, 1, 1, 2, 2, 2, 3, 3, 4], dtype=numpy.int32)
+    >>> dsts = cupy.asarray([1, 3, 4, 0, 1, 3, 4, 5, 5], dtype=numpy.int32)
+    >>> weights = cupy.asarray(
+    ...     [0.1, 2.1, 1.1, 5.1, 3.1, 4.1, 7.2, 3.2, 6.1], dtype=numpy.float32)
+    >>> source_vertices = cupy.asarray([0, 1], dtype=numpy.int32)
+    >>> resource_handle = pylibcugraph.ResourceHandle()
+    >>> graph_props = pylibcugraph.GraphProperties(
+    ...     is_symmetric=False, is_multigraph=False)
+    >>> G = pylibcugraph.SGGraph(
+    ...     resource_handle, graph_props, srcs, dsts, weights,
+    ...     store_transposed=False, renumber=False, do_expensive_check=False)
+    >>> (sources, destinations, edge_weights, subgraph_offsets) =
+    ...     pylibcugraph.ego_graph(resource_handle, G, source_vertices, 2, False)
+    # FIXME: update results
+    >>> sources
+    [0, 1, 1, 3, 1, 1, 3, 3, 4]
+    >>> destinations
+    [1, 3, 4, 4, 3, 4, 4, 5, 5]
+    >>> edge_weights
+    [0.1, 2.1, 1.1, 7.2, 2.1, 1.1, 7.2, 3.2, 6.1]
+    >>> subgraph_offsets
+    [0, 4, 9]
+
+    """
+
+    cdef cugraph_resource_handle_t* c_resource_handle_ptr = \
+        resource_handle.c_resource_handle_ptr
+    cdef cugraph_graph_t* c_graph_ptr = graph.c_graph_ptr
+    cdef cugraph_induced_subgraph_result_t* result_ptr
+    cdef cugraph_error_code_t error_code
+    cdef cugraph_error_t* error_ptr
+
+    cdef cugraph_type_erased_device_array_view_t* \
+        source_vertices_view_ptr = \
+            create_cugraph_type_erased_device_array_view_from_py_obj(
+                source_vertices)
+
+    error_code = cugraph_extract_ego(c_resource_handle_ptr,
+                                     c_graph_ptr,
+                                     source_vertices_view_ptr,
+                                     radius,
+                                     do_expensive_check,
+                                     &result_ptr,
+                                     &error_ptr)
+    assert_success(error_code, error_ptr, "cugraph_extract_ego")
+
+    # Extract individual device array pointers from result and copy to cupy
+    # arrays for returning.
+    cdef cugraph_type_erased_device_array_view_t* sources_ptr = \
+        cugraph_induced_subgraph_get_sources(result_ptr)
+    cdef cugraph_type_erased_device_array_view_t* destinations_ptr = \
+        cugraph_induced_subgraph_get_destinations(result_ptr)
+    cdef cugraph_type_erased_device_array_view_t* edge_weights_ptr = \
+        cugraph_induced_subgraph_get_edge_weights(result_ptr)
+    cdef cugraph_type_erased_device_array_view_t* subgraph_offsets_ptr = \
+        cugraph_induced_subgraph_get_subgraph_offsets(result_ptr)
+
+    cupy_sources = copy_to_cupy_array(
```

Comment:
Can you add a FIXME here to have this use the technique [here](https://github.com/rapidsai/cugraph/blob/93b9af2726290f8d38997942d9176b216dc351f5/python/pylibcugraph/pylibcugraph/uniform_neighbor_sample.pyx#L146) instead of a copy?